app/models/g5_authenticatable/user.rb

```diff
@@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
     end
 
     def clients
-      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
+      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
```

Comment:
So should I create a different policy?

On Tue, Feb 16, 2016 at 12:17 PM, Jonathan Samples <jonnysamps@gmail.com>
wrote:

> It is being used by the g5_header to list all of the clients that a user
> can choose from... that list needs to include the clients that would be
> implied by their location roles
>
> On Tue, Feb 16, 2016 at 12:10 PM, Maeve Revels <notifications@github.com>
> wrote:
>
>> In app/models/g5_authenticatable/user.rb
>> <https://github.com/G5/g5_authenticatable/pull/46#discussion_r53070594>:
>>
>> > @@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
>> >      end
>> >
>> >      def clients
>> > -      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
>> > +      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
>>
>> In terms of data-level authorization across G5 apps, I think @jonnysamps
>> <https://github.com/jonnysamps> is correct: a location-scoped role does
>> not imply the user has any permissions on the associated client (however, a
>> client-scoped role *does* imply the user has the same permissions on
>> associated locations). I think we should keep that as the default behavior
>> of any pundit policies distributed with the auth libraries.
>>
>> That being said, @vgarro <https://github.com/vgarro> is also correct -
>> calling anything other than resolve on a pundit policy scope is kind of
>> a smell. I'm not sure that this new
>> clients_from_client_and_location_roles method really belongs on the
>> ClientPolicy scope, and whether we should modify the existing clients
>> method on the user model. In what context is this being used?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/G5/g5_authenticatable/pull/46/files#r53070594>.
>>
>
>
>
> --
> Jonathan
>



-- 
Jonathan